lib/model/compose.dart

```diff
@@ -101,18 +126,33 @@ String wrapWithBacktickFence({required String content, String? infoString}) {
   return resultBuffer.toString();
 }
 
-/// An @-mention, like @**Chris Bobbe|13313**.
+/// An @-mention of an individual user, like @**Chris Bobbe|13313**.
 ///
 /// To omit the user ID part ("|13313") whenever the name part is unambiguous,
 /// pass a Map of all users we know about. This means accepting a linear scan
 /// through all users; avoid it in performance-sensitive codepaths.
-String mention(User user, {bool silent = false, Map<int, User>? users}) {
+String userMention(User user, {bool silent = false, Map<int, User>? users}) {
   bool includeUserId = users == null
     || users.values.where((u) => u.fullName == user.fullName).take(2).length == 2;
 
   return '@${silent ? '_' : ''}**${user.fullName}${includeUserId ? '|${user.userId}' : ''}**';
 }
 
+/// An @-mention of all the users in a conversation, like @**channel**.
+String wildcardMention(WildcardMentionOption wildcardOption, {
+  required PerAccountStore store,
+}) {
+  final isChannelWildcardAvailable = store.account.zulipFeatureLevel >= 247; // TODO(server-9)
+  assert(isChannelWildcardAvailable || wildcardOption != WildcardMentionOption.channel);
+  final isTopicWildcardAvailable = store.account.zulipFeatureLevel >= 224; // TODO(sever-8)
+  assert(isTopicWildcardAvailable || wildcardOption != WildcardMentionOption.topic);
+
+  final name = wildcardOption == WildcardMentionOption.stream && isChannelWildcardAvailable
+    ? WildcardMentionOption.channel.canonicalString
+    : wildcardOption.canonicalString;
```

Comment:
All these equality comparisons to different enum values feel like they'd be clearer as one `switch` statement that lists all the values.

Could start with `String name = wildcardOption.canonicalString;` and then have the `switch` mutate it in one case, to avoid repeating the common version and have the exception stand out.